app/controllers/users_controller.rb

```diff
@@ -26,12 +26,23 @@ def logout
     render json: { message: 'successfully logged out' }, status: :ok
   end
 
-  def delete_current_user
-    unless @current_user.destroy
-      return render json: { message: @user.errors.full_messages },
+  def recover_account
+    user = User.find_by(email_id: user_params[:email_id])
+    return render json: { message: 'User not found' }, status: :not_found if user.nil?
+
+    unless user.restore
+      return render json: { message: 'Your account cannot be restored' },
                     status: :unprocessable_entity
     end
```

Comment:
Refactor your login. Only if the user is logged in, then they should be able to recover their account.